src/test/java/org/codice/usng/CoordinateSystemTranslatorTest.java

```diff
@@ -0,0 +1,1552 @@
+        assertEquals(0, (int) utmCoordinate.getNorthing());
+        assertEquals(31, utmCoordinate.getZoneNumber(), 0);
+    }
+
+    @Test
+    public void testGithubData() throws ParseException {
+        //Using test data from github issue
+        //washington monument
+        //should return lat 38.8895 long -77.0352
+        double lat = 38.8895;
+        double lon = -77.0352;
+        int utmNorthing = 4306483;
+        int utmEasting = 323486;
+        int zoneNum = 18;
+        UsngCoordinate usng = UsngCoordinate.parseUsngString("18S UJ 23487 06483");
+
+        UtmCoordinate utmCoordinate = coordinateSystemTranslator.toUtm(new LatLonCoordinate(lat, lon));
+
+        assertEquals(utmEasting, (int) utmCoordinate.getEasting());
+        assertEquals(utmNorthing, (int) utmCoordinate.getNorthing());
+        assertEquals(zoneNum, utmCoordinate.getZoneNumber());
+
+        utmCoordinate = new UtmCoordinate(zoneNum, utmEasting, utmNorthing);
+        LatLonCoordinate utmToLL = coordinateSystemTranslator.toLatLon(utmCoordinate);
+
+        assertEquals(Math.round(lat * 10000), Math.round(utmToLL.getLat() * 10000));
+        assertEquals(Math.round(lon * 10000), Math.round(utmToLL.getLon() * 10000));
+
+        assertEquals(usng,
+                coordinateSystemTranslator.toUsng(new LatLonCoordinate(lat, lon),
+                        CoordinatePrecision.ONE_METER));
+
+        LatLonCoordinate usngToLL = coordinateSystemTranslator.toLatLon(usng);
+
+        assertEquals(Math.round(lat * 10000), Math.round(usngToLL.getLat() * 10000));
+        assertEquals(Math.round(lon * 10000), Math.round(usngToLL.getLon() * 10000));
+
+        //white house
+        //should return lat 38.8977 lon -77.0366
+        lat = 38.8977;
+        lon = -77.0366;
+        utmNorthing = 4307395;
+        utmEasting = 323385;
+        zoneNum = 18;
+        usng = UsngCoordinate.parseUsngString("18S UJ 23386 07396");
+
+        utmCoordinate = coordinateSystemTranslator.toUtm(new LatLonCoordinate(lat, lon));
+
+        assertEquals(utmEasting, (int) utmCoordinate.getEasting());
+        assertEquals(utmNorthing, (int) utmCoordinate.getNorthing());
+        assertEquals(zoneNum, utmCoordinate.getZoneNumber());
+
+        utmCoordinate = new UtmCoordinate(zoneNum, utmEasting, utmNorthing);
+        utmToLL = coordinateSystemTranslator.toLatLon(utmCoordinate);
+
+        assertEquals(Math.round(lat * 10000), Math.round(utmToLL.getLat() * 10000));
+        assertEquals(Math.round(lon * 10000), Math.round(utmToLL.getLon() * 10000));
+
+        assertEquals(usng,
+                coordinateSystemTranslator.toUsng(new LatLonCoordinate(lat, lon),
+                        CoordinatePrecision.ONE_METER));
+
+        usngToLL = coordinateSystemTranslator.toLatLon(usng);
+
+        assertEquals(Math.round(lat * 10000), Math.round(usngToLL.getLat() * 10000));
+        assertEquals(Math.round(lon * 10000), Math.round(usngToLL.getLon() * 10000));
+
+        //mount everest'
```

Comment:
There are a lot of duplicated blocks of test code.  I would recommend extracting common private test methods to reduce boilerplate.